File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        if(nums.Length==0) return 0;
        int res = 1;
        var visit = new HashSet<int>();
        foreach(int num in nums) visit.Add(num);
        for(int i=0;i<nums.Length;i++){
            int seq = 1;
            if(!visit.Contains(nums[i]-1)){
                while(visit.Contains(nums[i]+1)){
                    seq++;
                    nums[i]++;
                }
            }
            res = Math.Max(res, seq);
            if(res>nums.Length/2) break;
        }
        return res;
    }
}

File: Data Structures & Algorithms/non-cyclical-number/submission-3.java
class Solution {
    public boolean isHappy(int n) {
       Set<Integer> visit = new HashSet();
       while(n!=1){
         int k = sumOfSquareOfDigits(n);
         if(visit.contains(k)) return false;
         visit.add(k);
         n = k;
       }
       return true;
    }

    public int sumOfSquareOfDigits(int n){
        int res = 0;
        while(n!=0){
            int d = n%10;
            res+=(d*d);
            n=n/10;
        }
        return res;
    }
}
